pkg/crypto/certificatemanager/certificate_manager.go

```diff
@@ -9,37 +9,65 @@ import (
 	"os"
 	"path/filepath"
 
+	"github.com/cilium/cilium/pkg/hive/cell"
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
 	"github.com/cilium/cilium/pkg/policy/api"
+
+	"github.com/spf13/pflag"
 )
 
-const (
-	caDefaultName      = "ca.crt"
-	publicDefaultName  = "tls.crt"
-	privateDefaultName = "tls.key"
+var Cell = cell.Module(
+	"certificate-manager",
+	"Provides TLS certificates and secrets",
+
+	cell.Provide(NewManager),
+
+	cell.Config(defaultManagerConfig),
 )
 
-type k8sClient interface {
-	GetSecrets(ctx context.Context, ns, name string) (map[string][]byte, error)
+type CertificateManager interface {
+	GetTLSContext(ctx context.Context, tlsCtx *api.TLSContext, ns string) (ca, public, private string, err error)
+}
+
+type SecretManager interface {
```

Comment:
Hmm, looks like we have no tests for this. Perhaps worth writing them as these seem fairly straightforward.

Also I wonder how frequently we're calling `GetSecrets`?t It's every time doing a GetSecrets() API call to the api-server... and furthermore I wonder how well we're handling temporary failures of said API call? Following the references of GetTLSContext leads to some scary places that seem to imply we get here often if we have L7 rules.

Ping @jrajahalme. Worth refactoring this to cache replies? We could also maintain an up-to-date store of secrets (k8s + file) on the side (`Resource[api.Secret]` from `CoreV1().Secrets("")` though maybe all of it is too much?) and never do actual api-server or file system operations on the hot path.

Reply:
Yes, good concerns. That also makes me wonder if we currently deal with [changes](https://kubernetes.io/docs/concepts/configuration/secret/#editing-a-secret) to these secrets. You would think that if the secret updates, we should also update the rules that use them.

Reply:
At what point do we just want a controller watching for changes on secrets we use (not sure when the overhead of that is worth the simple approach)

Reply:
Also, if we do have to call out with a Get to apiserver, I wonder why we are not writing out the secret to the file (there's also the question of race conditions, but I'm not exactly sure where this file gets written to initially...)